statsmodels/survey/summary_stats.py

```diff
@@ -0,0 +1,480 @@
+        The weight for each observation. If none, an array 
+        of ones is constructed
+    nest : boolean
+        allows user to specify if PSU's with the same
+        PSU number in different strata are treated as distinct PSUs.
+
+    Attributes
+    ----------
+    self.weights : (n, ) array
+        The weight for each observation
+    self.nstrat : integer
+        The number of district strata
+    self.sclust : (n, ) array
+        The relabeled cluster array from 0, 1, ..
+    self.strat : (n, ) array
+        The related strata array from 0, 1, ...
+    self.ncs : (self.nstrat, ) array
+        Holds the number of clusters in each stratum
+    self.sfclust : ndarray
+        The stratum for each cluster
+    self.nclust : integer
+        The total number of clusters across strata
+    """
+    
+    def __init__(self, strata=None, cluster=None, weights=None, nest=True):
+        strata, cluster, self.weights = self._check_args(strata, cluster, weights)
+        self.weights = weights
+
+        # Recode strata and clusters as integer values 0, 1, ...                                                                                             
+        _, self.strat = np.unique(strata, return_inverse=True)
+        _, clust = np.unique(cluster, return_inverse=True)
+
+        # the number of distinct strata
+        self.nstrat = max(self.strat) + 1
+
+        # If requested, recode the PSUs to be sure that the same PSU number in                                                                               
+        # different strata are treated as distinct PSUs.  This is the same as                                                                                
+        # the nest option in R.                                                                                                                              
+        if nest:
+            m = max(clust) + 1
+            sclust = clust + m*self.strat
+            _, self.sclust = np.unique(sclust, return_inverse=True)
+        else:
+            self.sclust = clust.copy()
+            
+        # The number of clusters per stratum                                                                                                                 
+        _, ii = np.unique(self.sclust, return_index=True)
+        self.ncs = np.bincount(self.strat[ii])
+
+        # The stratum for each cluster                                                                                                                       
+        _, ii = np.unique(self.sclust, return_index=True)
+        self.sfclust = self.strat[ii]
+
+        # The total number of clusters over all stratum
+        self.nclust = np.sum(self.ncs)
+
+    def summary(self):
+        """
+        The __str__ method for our data
+        """
+        print("Number of observations:", len(self.strat))
+        print("Sum of weights:", self.weights.sum())
```

Comment:
don't use `print` just build and return the string. e.g. with "\n".join(list_of_lines)